Classes/Emogrifier.php

```diff
@@ -445,19 +428,21 @@ protected function process(\DOMDocument $xmlDocument)
      */
     private function mapAllInlineStylesToHtmlAttributes(\DOMXPath $xPath)
     {
-        $nodesWithStyleAttributes = $xPath->query('//*[@style]');
-        if ($nodesWithStyleAttributes === false) {
+        try {
+            $nodesWithStyleAttribute = $this->getAllNodesWithStyleAttribute($xPath);
+        } catch (\RuntimeException $e) {
             if ($this->debug) {
                 throw new \RuntimeException(
-                    'Possibly malformed DOMXPath query expression or invalid DOMXPath object.',
-                    1508437884
+                    $e->getMessage(),
+                    1508437884,
```

Comment:
We can just rethrow `$e` here, I think.

Oh, now that I think about this, this might result in the same exception code used in several places. How can we solve this in an elegant way?

Reply:
Same exception code is why I chose to create a new exception object and specify the original as previous in the exception chain (the third parameter) instead of simply re-throwing $e. This way, a distinct exception code can be defined in every place, where `getAllNodesWithStyleAttribute()` is being used.

Reply:
Ah, makes sense. Thanks for the explanation!

Reply:
The point of the exception code is to help find the place where the exception originally has been thrown. So I don't think we need to a different code here. And to find out the place where the exception gets re-thrown, the stack trace of the exception can be used. This will help keep the code simple. What do you think?